cleanlab/datalab/internal/issue_manager/identifier_column.py

```diff
@@ -0,0 +1,110 @@
+        min_val, max_val = unique_sorted[0], unique_sorted[-1]
+        expected_range = np.arange(min_val, max_val + 1)
+        if expected_range.size == 1 or unique_sorted.size != expected_range.size:
+            return False
+        return bool((expected_range == unique_sorted).all())
+
+    def _prepare_features(
+        self, features: Optional[Union[npt.NDArray, pd.DataFrame, list, dict]]
+    ) -> npt.NDArray:
+        """
+        Prepare the features for issue check.
+
+        Args:
+            features: The input features.
+
+        Returns:
+            features: features as npt.NDArray
+        """
+        if isinstance(features, np.ndarray):
+            return features.T
+        elif isinstance(features, pd.DataFrame) or isinstance(features, dict):
+            return np.array([col for _, col in features.items()])
+        elif isinstance(features, list):
+            for col_list in features:
+                if not isinstance(col_list, list):
+                    raise ValueError("features must be a list of lists if it features is a list.")
+            return np.array([np.array(col_list) for col_list in features])
+        else:
+            raise ValueError(
+                "features must be a numpy array or a pandas DataFrame. or list\
+                    or dict that can be converted to a numpy array."
+            )
+
+    def find_issues(
+        self, features: Optional[Union[npt.NDArray, pd.DataFrame, list, dict]], **kwargs
+    ) -> None:
+        """
+        Find identifier columns in the given dataset.
+
+        Parameters:
+            features (Optional[npt.NDArray | pd.DataFrame | list | dict]):
+            The dataset to check for identifier columns.
+        Returns:
+            None
+        """
+        if features is None:
+            raise ValueError("features must be provided to check for identifier columns.")
+        processed_features = self._prepare_features(features)
+
+        scores = np.array(
+            [
+                np.issubdtype(feature.dtype, np.integer) and self._is_sequential(feature)
+                for feature in processed_features
+            ]
+        )
+        issue_indices = np.where(scores)
+        # this issue does not reflect rows at all so we set the score to 1.0 for all rows in the issue attribute
+        # and set the is_identifier_column_issue to False
+        self.issues = pd.DataFrame(
+            {
+                f"is_{self.issue_name}_issue": False,
+                self.issue_score_key: [1.0 for _ in range(processed_features.shape[1])],
+            },
+        )
+        # score in summary should be 1.0 if the issue is not presend and 0.0 if at least one column is an identifier column
+        self.summary = self.make_summary(score=1.0 if scores.sum() == 0 else 0.0)
+        self.info = {
+            "identifier_columns": issue_indices,
+            "num_identifier_columns": scores.sum(),
```

Comment:
yes, good catch, changed it

Reply:
Thanks. Is there any reason for still using the `sum` function instead of the `size` function? Also, as suggested before, please rename `issue_indices` to `identifier_column_indices` for more clarity ([reference comment](https://github.com/cleanlab/cleanlab/pull/1120#discussion_r1748878263)).

Reply:
done